File: src/backend/shared/geo.py
"""Lightweight geo helpers for bbox filtering."""
from __future__ import annotations

import math


def in_bbox(
    lat: float,
    lon: float,
    min_lat: float,
    min_lon: float,
    max_lat: float,
    max_lon: float,
) -> bool:
    return min_lat <= lat <= max_lat and min_lon <= lon <= max_lon


def radius_km_from_height_m(height_m: float) -> float:
    """Approximate ground search radius from camera altitude."""
    h = max(100.0, float(height_m))
    return min(80.0, max(2.0, h / 800.0))


def distance_nm_from_height_m(height_m: float) -> int:
    km = radius_km_from_height_m(height_m)
    nm = km / 1.852
    return int(max(50, min(400, round(nm))))

File: src/frontend/geo.js
/** Client-side geo helpers (mirror backend/shared/geo.py). */

export function radius_km_from_height_m(heightM) {
  const h = Math.max(100, Number(heightM) || 12000);
  return Math.min(80, Math.max(2, h / 800));
}

export function distance_nm_from_height_m(heightM) {
  const km = radius_km_from_height_m(heightM);
  const nm = km / 1.852;
  return Math.max(50, Math.min(400, Math.round(nm)));
}

export function bboxQueryParams(bbox) {
  if (!bbox) return "";
  const q = new URLSearchParams({
    min_lat: String(bbox.min_lat),
    min_lon: String(bbox.min_lon),
    max_lat: String(bbox.max_lat),
    max_lon: String(bbox.max_lon),
  });
  return `&${q.toString()}`;
}
